clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -458,8 +312,8 @@ protected Map<Node, FetchSessionHandler.FetchRequestData> prepareFetchRequests()
             // Use the preferred read replica if set, otherwise the partition's leader
             Node node = selectReadReplica(partition, leaderOpt.get(), currentTimeMs);
 
-            if (client.isUnavailable(node)) {
-                client.maybeThrowAuthFailure(node);
+            if (nodeStatusDetector.isUnavailable(node)) {
```

Comment:
Just for the record, if we agree on this NodeStatusDetector component, I would integrate a similar logic to the ListOffsetRequestManager for checking leader status before sending a request. lgtm